modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/PlannerPhase.java

```diff
@@ -73,6 +76,8 @@
                 ProjectConverterRule.INSTANCE,
                 ProjectTraitsPropagationRule.INSTANCE,
                 FilterConverterRule.INSTANCE,
+                FilterMergeRule.INSTANCE,
```

Comment:
It looks like standard Calcite rule uses `Filter.class` as its operand. But I think it is much more efficient to use `LogicalFilter.class` operands instead. Can you rewrite standard rule and make it use logical operands instead of both physical and logical ones?